src/controls/sequence_node.cpp

```diff
@@ -47,6 +47,7 @@ NodeStatus SequenceNode::tick()
       case NodeStatus::FAILURE: {
         // Reset on failure
         resetChildren();
+        propagateGeneralStatusFromFailingChild(current_child_node);
```

Comment:
There are instances where your order is

1.  reset
2. propagate

and others where its the opposite (1. propagate, 2. reset). Which makes more sense? Or is `resetChildren` independent of our general status?

Reply:
`reserChildren` is independent of General status implementation - it is responsible to halt nodes after execution. 
The order here do not matters, but I'll change it to propagate->reset for consistency